relayer/application_relayer.go

```diff
@@ -276,53 +271,36 @@ func (r *ApplicationRelayer) createSignedMessage(
 		signedWarpMessageBytes hexutil.Bytes
 		err                    error
 	)
-	for attempt := 1; attempt <= maxRelayerQueryAttempts; attempt++ {
-		r.logger.Debug(
-			"Relayer collecting signatures from peers.",
-			zap.Int("attempt", attempt),
-			zap.String("sourceBlockchainID", r.sourceBlockchain.GetBlockchainID().String()),
-			zap.String("destinationBlockchainID", r.relayerID.DestinationBlockchainID.String()),
-			zap.String("signingSubnetID", r.signingSubnetID.String()),
-		)
-
-		err = r.sourceWarpSignatureClient.CallContext(
-			context.Background(),
-			&signedWarpMessageBytes,
-			"warp_getMessageAggregateSignature",
-			unsignedMessage.ID(),
-			r.warpConfig.QuorumNumerator,
-			r.signingSubnetID.String(),
-		)
-		if err == nil {
-			warpMsg, err := avalancheWarp.ParseMessage(signedWarpMessageBytes)
-			if err != nil {
-				r.logger.Error(
-					"Failed to parse signed warp message",
-					zap.Error(err),
-				)
-				return nil, err
-			}
-			return warpMsg, err
-		}
-		r.logger.Info(
-			"Failed to get aggregate signature from node endpoint. Retrying.",
-			zap.Int("attempt", attempt),
-			zap.Error(err),
-		)
-		if attempt != maxRelayerQueryAttempts {
-			// Sleep such that all retries are uniformly spread across totalRelayerQueryPeriodMs
-			// TODO: We may want to consider an exponential back off rather than a uniform sleep period.
-			time.Sleep(time.Duration(signatureRequestRetryWaitPeriodMs/maxRelayerQueryAttempts) * time.Millisecond)
-		}
-	}
-	r.logger.Warn(
-		"Failed to get aggregate signature from node endpoint",
-		zap.Int("attempts", maxRelayerQueryAttempts),
+	err = utils.WithMaxRetriesLog(
```

Comment:
Just a head's up, the Warp API integration is planned to be deprecated in the near future. It's still reasonable to integrate exponential backoff here for the time being.